### ISSUE TITLE
_remote_checksum - host not found - due to usage of conn.host (with ip) instead of hostname

### ISSUE THREAD

**Issue by queria** [ansible/ansible#9543]
- using ansible from devel branch (7812c70d3b)
- having custom name "some-machine" for a host in hosts file
- and exact ip as ansible_ssh_host in corresponding host_vars/some-machine

for ex. copy action fails on:

```
fatal: [some-machine] => Traceback (most recent call last):
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/__init__.py", line 583, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/__init__.py", line 757, in _executor_internal
    return self._executor_internal_inner(host, self.module_name, self.module_args, inject, port, complex_args=complex_args)
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/__init__.py", line 990, in _executor_internal_inner
    result = handler.run(conn, tmp, module_name, module_args, inject, complex_args)
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/action_plugins/copy.py", line 179, in run
    remote_checksum = self.runner._remote_checksum(conn, tmp_path, dest_file)
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/__init__.py", line 1167, in _remote_checksum
    inject = self.get_inject_vars(conn.host)
  File "/home/psedlak/all/src/ansible/lib/ansible/runner/__init__.py", line 607, in get_inject_vars
    host_variables = self.inventory.get_variables(host, vault_password=self.vault_pass)
  File "/home/psedlak/all/src/ansible/lib/ansible/inventory/__init__.py", line 443, in get_variables
    raise Exception("host not found: %s" % hostname)
Exception: host not found: 10.10.10.1
```

My guess after quickly looking into the code is that:
- _copy.run_ calls _runner._remote_checksum_ (btw calling (single) underscore method from outside?) passing conn (it does not have hostname avail)
- __remote_checksum_ uses _inventory.get_..._ which expects _hostname_ but from _conn_ object it gets "just" _ip_ to where connection was initialized
- that results in not finding _ip_ value in the _inventory_ (_hosts_cache there has _hostnames_ as keys)

I'm not sure what should be correct behaviour where here. Have inventory use ip's for lookup/caching? There seems to be also conn.delegate which holds the hostname so maybe using that value (if it exists, otherwise continue with .host as it is atm)?

I didn't checked but it seems this was "introduced" (as side effect) by change f1267c0b "Move from md5 to sha1..." when remote_md5 was changed to remote_checksum and instead of doing plain conn.shell.md5 it started consulting HostVars etc, and based on the size of the change I assume it may impact also assemble, fetch, template and unarchive actions.

**Comment by queria** [comment]
Now I've pulled new changes and it seems this is already fixed by 30c50020, sorry for not noticing it before.

**Comment by queria** [comment]
act. it would be also Duplicate of issue #9528
